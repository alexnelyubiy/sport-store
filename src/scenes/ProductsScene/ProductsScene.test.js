import React from 'react';
import { shallow } from 'enzyme';
import ProductScene from './ProductsScene';

const testProps = {
  products: [{ name: 'Item 1', price: 290, id: 1 }, { name: 'Item 2', price: 20, id: 2 }],
  cartProducts: [{ name: 'Item 1', price: 290, id: 1 }, { name: 'Item 2', price: 20, id: 2 }],
};
describe('Given the ProductScene component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeEach(() => {
      component = shallow(<ProductScene {...testProps} />);
    });

    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
//
