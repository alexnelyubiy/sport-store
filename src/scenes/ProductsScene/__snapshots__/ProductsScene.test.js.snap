// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ProductScene component when the component is rendered should render the component with provided props 1`] = `
<Master
  hasCart={true}
  itemsCount={2}
  title="Products"
>
  <ProductsScenestyled__ProductsWrapper>
    <ProductContainer
      key="1"
      product={
        Object {
          "id": 1,
          "name": "Item 1",
          "price": 290,
        }
      }
    />
    <ProductContainer
      key="2"
      product={
        Object {
          "id": 2,
          "name": "Item 2",
          "price": 20,
        }
      }
    />
  </ProductsScenestyled__ProductsWrapper>
</Master>
`;
